framework/src/st_things/things_stack/things_iotivity_lock.c

```diff
@@ -0,0 +1,75 @@
+/****************************************************************************
+ *
+ * Copyright 2018 Samsung Electronics All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied. See the License for the specific
+ * language governing permissions and limitations under the License.
+ *
+ ****************************************************************************/
+
+#include "things_iotivity_lock.h"
+#include "logging/things_logger.h"
+
+#define TAG "[things_iotivity_api_lock]"
+
+static pthread_mutex_t g_iotivity_lock;
```

Comment:
please include <sys/types.h>

Reply:
Sure. But it works even without this header. Do you know how?